### ISSUE TITLE
0.2.16 build fails on linux

### ISSUE THREAD

**Issue by lgbaldoni** [xtingray/tupitube.desk#57]
Am I missing something here?
```
> ./configure
Traceback (most recent call last):
        4: from configure.rb:42:in `<main>'
        3: from configure.rb:42:in `require_relative'
        2: from /home/abuild/rpmbuild/BUILD/tupitube.desk/qonf/configure.rb:38:in `<top (required)>'
        1: from /usr/lib64/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
/usr/lib64/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require': cannot load such file -- os (LoadError)
```

**Comment by xtingray** [comment]
Yes, you are missing the Python dependency "os". 
You should fix it running: `sudo gem install so`

**Comment by lgbaldoni** [comment]
Ok, I've managed to get rubygem-os accepted in the relevant openSUSE repository. The trouble is now with quazip 1.x.

As you can see from the [log](https://github.com/xtingray/tupitube.desk/files/5483200/tupitube0216.txt), quazip is not being found in any of the hardcoded search paths because the include dir is now versioned (/usr/include/QuaZip-Qt5-1.1 in this case), but I couldn't understand where to patch to search for it via pkgconfig.

Speaking of which and as dispassionate suggestion, why use a mix of ruby and qmake instead of a cmake build script?

**Comment by xtingray** [comment]
The migration of the current building system to CMake is part of my huge To-Do list. Why haven't done it yet? Well, this project started 10 years ago and along all this time, only three or four guys have been curious about the source code of the project (including you). The 99.9% of people interested in using this project only cares about getting the binaries I created, nothing more. 

Understanding that I am absolutely alone on this, I focused on developing the features that the users are requesting all the time.
About the packaging of TupiTube for Linux systems, the challenge from my point of view is related to the Qt installation, as every Linux distro splits the Qt libs in different components, using different names. I stopped to fight against that a long time ago, so I package my own set of Qt libs as part of the Linux binary installer I release. If you think that it's possible to fix the Qt compatibility between Linux distros using CMake, be my guest. 

**Note:** Some guy helped me once to make the Debian package for TupiTube a long time ago. In some way, the Debian community packages my project, but as far as I know, they don't use CMake, instead they made a work-around using my config script: https://packages.debian.org/search?keywords=tupi

Finally, how I deal with the building process of TupiTube on my own Linux platform? Well, I follow this procedure from time to time to update libraries and dependencies and that's it: https://www.maefloresta.com/wiki/doku.php?id=ubuntu_compilation_howto

**Note:** I had to modify some files on the Quazip source files, so I maintain my own branch of that library.
